core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java

```diff
@@ -400,11 +401,19 @@ public DetailedEnvironmentV4Response registerExternalDatalake(String environment
 
     @Override
     protected void prepareDeletion(Environment environment) {
-        Long aliveStacks = stackService.countAliveByEnvironment(environment);
-        Long aliveClusters = clusterService.countAliveByEnvironment(environment);
-        if (aliveStacks > 0 || aliveClusters > 0) {
+        List<String> aliveStacks = stackService.getNameOfAliveByEnvironment(environment);
+        if (!aliveStacks.isEmpty()) {
             throw new BadRequestException("Cannot delete environment. "
-                    + "All clusters must be terminated before environment deletion. Alive clusters: " + aliveStacks);
+                    + "All clusters must be terminated before environment deletion. Alive cluster(s): "
+                    + String.join(", ", aliveStacks) + "."
+            );
+        }
+        List<String> aliveClusters = clusterService.getNameOfAliveByEnvironment(environment);
+        if (!aliveClusters.isEmpty()) {
+            throw new BadRequestException("Cannot delete environment. "
+                    + "All clusters must be terminated before environment deletion. One or more clusters are in bad state after termination: "
```

Comment:
How often does it happen that the set of alive clusters and alive stacks does not match?